app/Http/Controllers/Admin/CategoriesController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Models\Meeting;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Storage;
+
+class CategoriesController extends Controller
+{
+    private $category;
+    private $meeting;
+    const ICON_FOLDER = 'image/category/';
+
+    public function __construct(Category $category, Meeting $meeting)
+    {
+        $this->category = $category;
+        $this->meeting = $meeting;
+    }
+    public function index()
+    {
+        $all_categories = $this->category->withTrashed()->paginate(10);
+        return view('admin.chatrooms.categories.index')
+            ->with('all_categories', $all_categories);
+    }
+    public function show($id)
+    {
+        // update Meetings Status
+        $this->meeting->updateStatus();
+        
+        $category = $this->category->withTrashed()->findOrFail($id);
+        $meetings = $category->meetings()->withTrashed()->latest()->paginate(10);
+        return view('admin.chatrooms.categories.show')
+            ->with('category', $category)
+            ->with('meetings', $meetings)
+            ->with('statusColor', $this->meeting->statusColor());
+    }
+    public function add()
+    {
+        return view('admin.chatrooms.categories.add');
+    }
+    public function edit($id)
+    {
+        $category = $this->category->withTrashed()->findOrFail($id);
+        return view('admin.chatrooms.categories.edit')
+        ->with('category', $category);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name'          => 'required|min:1|max:20|unique:categories,name',
+            'color'         => 'required|regex:/^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$/', //Hex Color
+            'icon'          => 'required|mimes:jpg,jpeg,png,gif',
+            'description'   => 'required|min:5|max:100'
+        ]);
+
+        $this->category->name = str_replace(" ", "_", ucwords($request->name));
+        $this->category->color = $request->color;
+        $this->category->description = $request->description;
+        $this->category->icon = $this->saveIcon($request);
+        $this->category->save();
+        return redirect()->route('admin.chatroom.category.index');
+    }
+    public function update(Request $request, $id)
+    {
+        $category = $this->category->withTrashed()->findOrFail($id);
+        $request->validate([
+            'name' => 'required|min:1|max:20|unique:categories,name,' . $category->id,
+            'color'=> 'required|regex:/^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$/',
+            'icon' => 'mimes:jpg,jpeg,png,gif',
+            'description' => 'required|min:5|max:100'
+        ]);
+        
+        $category->name = str_replace(" ", "_", ucwords(strtolower($request->name)));
```

Comment:
For category name, maybe we should keep the value as is and maybe create a new column like `slug` that holds the value with underscore(`_`).

Reply:
it's still the same